apps/cloudinary/src/index.js

```diff
@@ -100,12 +109,28 @@ async function openDialog(sdk, currentValue, config) {
   });
 
   if (result && Array.isArray(result.assets)) {
-    return result.assets.map(asset => pick(asset, FIELDS_TO_PERSIST));
+    return result.assets.map(asset => extractAsset(asset));
   } else {
     return [];
   }
 }
 
+function extractAsset(asset){
+  let res = pick(asset,FIELDS_TO_PERSIST);
+  // if we have a derived images, we replace the URL with the derived URL and store the origianl URL seperatly
+  if(asset.derived){
+    res = {
+      ...res,
+      original_url: res.url,
+      original_secure_url: res.secure_url,
+      url: asset.derived[0].url,
+      secure_url: asset.derived[0].secure_url,
+      raw_transformation: asset.derived[0].raw_transformation
```

Comment:
Hey @asisayag2 just one last question. I was looking over this code one last time to ensure that this will be backwards compatible for users that already have the app installed. What exactly is the `derived` asset doing here and do you think it will somehow affect current users?

Just for context, when we ship this, all the current users will get this change. If this somehow alters the data in their entries, it may break their front-ends.

Reply:
Hey @davidfateh , 
When a user is selecting an image from our media library widget, you get a json back with the asset's info. 
If the asset has any transformation on it (and once we add the default f_auto/q_auto, you get this by default), there's another section in the returned json, describing the transformed image (we call it derived) 

If there is a derived section, we extract both the original image's URL and the derived URL 

As I'm still using the URL field, I don't see a reason that this will break BC